### DISCUSSION TITLE
Potentially should be constant/view/pure but is not error in the code

### DISCUSSION THREAD

**Discussion by ShivSankar** [smartcontractkit/full-blockchain-solidity-course-py#1802] [Q&A] [answered]
I'm getting the warning which is highlighted in yellowish color when I' m using this code. I understand that the issue in addPeople function but can anyone tell me how to get rid of it?

![simpleStorageIssue1](https://user-images.githubusercontent.com/12861776/205655848-b26154a9-32e7-46f4-9609-7bbd83439f13.PNG)

**Comment by cromewar** [comment, marked as answer]
I checked this and there is nothing wrong with your code, this is just a recommendation provided by the solidity extension on REMIX but you can safely ignore this warning, you cannot use those types because:

* Pure is for mathematical operations.
* constant for functions which will not alter the state of a contract.
* view will not make state changes and this function does.

I hope this info might help, cheers.
